### PR TITLE
API Responsiveness: Add @immediateServie to submit(job)

### PR DESCRIPTION
**Pull request by Mykhailenko**
[trello ](https://trello.com/c/SNBYK90S/2983-benchmark-azure-make-api-responsiveness-1s)

This is second PR (after [PR3156](https://github.com/ow2-proactive/scheduling/commit/babae684040dc9dfa9547c37207d1ec69e0449fe)). 

Making submit method ImmediateService will un-freeze SchedulerFrontend - it will be available for other requests. However, this PR wont make submit method faster.

### CODE REVIEW

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java [hunk 1]
```diff
@@ -420,6 +420,7 @@ public void connect(UniqueID sourceBodyID, UserIdentificationImpl identification
      * {@inheritDoc}
      */
     @Override
+    @ImmediateService
     public JobId submit(Job userJob)
             throws NotConnectedException, PermissionException, SubmissionClosedException, JobCreationException {
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java]: MINOR Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 71 issues
* ![MINOR][MINOR] 71 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [SchedulerFrontend.java#L457](https://github.com/ow2-proactive/scheduling/blob/529d1632b7787c86ace9da1f53b7e9b595d831fb/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java#L457): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [SchedulerFrontend.java#L467](https://github.com/ow2-proactive/scheduling/blob/529d1632b7787c86ace9da1f53b7e9b595d831fb/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java#L467): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [SchedulerFrontend.java#L478](https://github.com/ow2-proactive/scheduling/blob/529d1632b7787c86ace9da1f53b7e9b595d831fb/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java#L478): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [SchedulerFrontend.java#L513](https://github.com/ow2-proactive/scheduling/blob/529d1632b7787c86ace9da1f53b7e9b595d831fb/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java#L513): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [SchedulerFrontend.java#L523](https://github.com/ow2-proactive/scheduling/blob/529d1632b7787c86ace9da1f53b7e9b595d831fb/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java#L523): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [SchedulerFrontend.java#L533](https://github.com/ow2-proactive/scheduling/blob/529d1632b7787c86ace9da1f53b7e9b595d831fb/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java#L533): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [SchedulerFrontend.java#L540](https://github.com/ow2-proactive/scheduling/blob/529d1632b7787c86ace9da1f53b7e9b595d831fb/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java#L540): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [SchedulerFrontend.java#L562](https://github.com/ow2-proactive/scheduling/blob/529d1632b7787c86ace9da1f53b7e9b595d831fb/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java#L562): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [SchedulerFrontend.java#L572](https://github.com/ow2-proactive/scheduling/blob/529d1632b7787c86ace9da1f53b7e9b595d831fb/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java#L572): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [SchedulerFrontend.java#L582](https://github.com/ow2-proactive/scheduling/blob/529d1632b7787c86ace9da1f53b7e9b595d831fb/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java#L582): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by Mykhailenko** [comment]
Retest this please

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 71 issues
* ![MINOR][MINOR] 71 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [SchedulerFrontend.java#L457](https://github.com/ow2-proactive/scheduling/blob/6fbcbf1c7e24205eb7598a2672af43c7138f24c2/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java#L457): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [SchedulerFrontend.java#L467](https://github.com/ow2-proactive/scheduling/blob/6fbcbf1c7e24205eb7598a2672af43c7138f24c2/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java#L467): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [SchedulerFrontend.java#L478](https://github.com/ow2-proactive/scheduling/blob/6fbcbf1c7e24205eb7598a2672af43c7138f24c2/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java#L478): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [SchedulerFrontend.java#L513](https://github.com/ow2-proactive/scheduling/blob/6fbcbf1c7e24205eb7598a2672af43c7138f24c2/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java#L513): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [SchedulerFrontend.java#L523](https://github.com/ow2-proactive/scheduling/blob/6fbcbf1c7e24205eb7598a2672af43c7138f24c2/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java#L523): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [SchedulerFrontend.java#L533](https://github.com/ow2-proactive/scheduling/blob/6fbcbf1c7e24205eb7598a2672af43c7138f24c2/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java#L533): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [SchedulerFrontend.java#L540](https://github.com/ow2-proactive/scheduling/blob/6fbcbf1c7e24205eb7598a2672af43c7138f24c2/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java#L540): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [SchedulerFrontend.java#L562](https://github.com/ow2-proactive/scheduling/blob/6fbcbf1c7e24205eb7598a2672af43c7138f24c2/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java#L562): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [SchedulerFrontend.java#L572](https://github.com/ow2-proactive/scheduling/blob/6fbcbf1c7e24205eb7598a2672af43c7138f24c2/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java#L572): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [SchedulerFrontend.java#L582](https://github.com/ow2-proactive/scheduling/blob/6fbcbf1c7e24205eb7598a2672af43c7138f24c2/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java#L582): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 71 issues
* ![MINOR][MINOR] 71 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [SchedulerFrontend.java#L457](https://github.com/ow2-proactive/scheduling/blob/70c9dfac16833fd0ee717dedf3c8b5a0be9fc39b/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java#L457): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [SchedulerFrontend.java#L467](https://github.com/ow2-proactive/scheduling/blob/70c9dfac16833fd0ee717dedf3c8b5a0be9fc39b/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java#L467): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [SchedulerFrontend.java#L478](https://github.com/ow2-proactive/scheduling/blob/70c9dfac16833fd0ee717dedf3c8b5a0be9fc39b/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java#L478): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [SchedulerFrontend.java#L513](https://github.com/ow2-proactive/scheduling/blob/70c9dfac16833fd0ee717dedf3c8b5a0be9fc39b/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java#L513): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [SchedulerFrontend.java#L523](https://github.com/ow2-proactive/scheduling/blob/70c9dfac16833fd0ee717dedf3c8b5a0be9fc39b/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java#L523): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [SchedulerFrontend.java#L533](https://github.com/ow2-proactive/scheduling/blob/70c9dfac16833fd0ee717dedf3c8b5a0be9fc39b/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java#L533): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [SchedulerFrontend.java#L540](https://github.com/ow2-proactive/scheduling/blob/70c9dfac16833fd0ee717dedf3c8b5a0be9fc39b/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java#L540): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [SchedulerFrontend.java#L562](https://github.com/ow2-proactive/scheduling/blob/70c9dfac16833fd0ee717dedf3c8b5a0be9fc39b/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java#L562): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [SchedulerFrontend.java#L572](https://github.com/ow2-proactive/scheduling/blob/70c9dfac16833fd0ee717dedf3c8b5a0be9fc39b/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java#L572): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [SchedulerFrontend.java#L582](https://github.com/ow2-proactive/scheduling/blob/70c9dfac16833fd0ee717dedf3c8b5a0be9fc39b/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java#L582): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**